src/app/search/search.component.ts

```diff
@@ -11,132 +11,265 @@ import { StoryModel, SeriesModel } from '../shared';
 
 export class SearchComponent implements OnInit {
 
+  static TAB_STORIES = 'stories';
+  static TAB_SERIES = 'series';
   selectedTab: string;
 
   isLoaded = false;
   totalCount: number;
   pages: number[];
-  noStories: boolean;
+  noResults: boolean;
   auth: HalDoc;
-  stories: StoryModel[];
-  storyLoaders: boolean[];
+  storiesResults: StoryModel[];
+  seriesResults: SeriesModel[];
+  loaders: boolean[];
 
   currentPage: number;
   showNumPages: number = 10;
   pagesBegin: number;
   pagesEnd: number;
 
-  searchText: string;
-  searchSeriesId: number;
-  searchSeries: SeriesModel;
+  searchTextParam: string;
+  searchGenreParam: string;
+  searchSubGenreParam: string;
+  searchSeriesIdParam: number;
+  searchSeriesParam: SeriesModel;
+  searchOrderByParam: string = 'updated_at';
+  searchOrderDescParam: boolean = true;
   allSeriesIds: number[];
   allSeries: any;
 
-  searchOrderBy: string;
-  orderByOptions: any[] = [
+  orderByOptionsStories: any[] = [
     {
-      id: 'TITLE',
+      id: 'title',
       name: 'Story Title'
     },
     {
-      id: 'UPDATED',
+      id: 'updated_at',
       name: 'Last Updated'
     },
     {
-      id: 'PUBLISHED',
+      id: 'published_at',
       name: 'When Published'
     }
   ];
 
+  orderByOptionsSeries: any[] = [
+    {
+      id: 'title',
+      name: 'Series Title'
+    },
+    {
+      id: 'updated_at',
+      name: 'Last Updated'
+    }
+  ];
+
   constructor(private cms: CmsService,
+              private router: Router,
               private route: ActivatedRoute) {}
 
   ngOnInit() {
-    this.route.params.subscribe(params => {
-      this.selectedTab = params['tab'];
-      this.searchSeriesId = +params['seriesId'];
+    this.route.params.forEach((params) => {
+      this.selectedTab = params['tab'] || SearchComponent.TAB_STORIES;
+      this.searchSeriesIdParam = +params['seriesId'];
 
       this.cms.follow('prx:authorization').subscribe((auth) => {
         this.auth = auth;
+        if (this.selectedTab === SearchComponent.TAB_STORIES) {
+          this.initStorySearch();
+        } else {
+          this.initSeriesSearch();
+        }
+      });
+    });
+  }
 
-        this.allSeriesIds = [-1];
-        this.allSeries = {};
-        auth.followItems('prx:series', {filters: 'v4'}).subscribe((series) => {
-          for (let s of <HalDoc[]> series) {
-            this.allSeriesIds.push(s.id);
-            this.allSeries[s.id] = new SeriesModel(auth, s, false);
-          }
+  initStorySearch() {
+    this.allSeriesIds = [-1];
+    this.allSeries = {};
+    this.auth.followItems('prx:series', {filters: 'v4'}).subscribe((series) => {
+      for (let s of <HalDoc[]> series) {
```

Comment:
Doesn't really hurt anything ... but I'm not sure you need this cast here.  `followItems` should already be returning a `Haldoc[]`.